src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -158,11 +168,178 @@ class SchemaService(private val options: Neo4jOptions, private val jobId: String
     struct
   }
 
+  def countForNodeWithQuery(filters: Array[Filter]): Long = {
+    val query = if (filters.isEmpty) {
+      options.nodeMetadata.labels
+        .map(_.quote())
+        .map(label => s"""
+             |MATCH (:$label)
+             |RETURN count(*) AS count""".stripMargin)
+        .mkString(" UNION ALL ")
+    } else {
+      queryReadStrategy.createStatementForNodeCount(options)
+    }
+    log.info(s"Executing the following counting query on Neo4j: $query")
+    session.run(query)
+      .list()
+      .asScala
+      .map(_.get("count"))
+      .map(count => if (count.isNull) 0L else count.asLong())
+      .min
+  }
+
+  def countForRelationshipWithQuery(filters: Array[Filter]): Long = {
+    val query = if (filters.isEmpty) {
+      val sourceQueries = options.relationshipMetadata.source.labels
```

Comment:
why can't we use apoc.meta.stats here?

Reply:
we use it, this is the fallback in case it's not present, look here:
https://github.com/neo4j-contrib/neo4j-spark-connector/pull/164/files#diff-85bfb2242b03b769ed6b534ff8992f88R255